File: app/src/main/java/com/cyder/atsushi/youtubesync/server/CreateRoom.java
package com.cyder.atsushi.youtubesync.server;

import com.cyder.atsushi.youtubesync.MainActivity;
import com.cyder.atsushi.youtubesync.R;
import com.cyder.atsushi.youtubesync.json_data.BaseResponse;

/**
 * Created by atsushi on 2017/10/27.
 */

public class CreateRoom extends HttpRequestsHelper {
    private CreateRoomInterface listener = null;

    public void setListener(CreateRoomInterface listener) {
        this.listener = listener;
    }

    public CreateRoom() {
        super(BaseResponse.class);
    }

    public void post(final String name, final String description){
        super.post(new com.cyder.atsushi.youtubesync.json_data.CreateRoom(name, description), "rooms", new HttpRequestCallback() {
            @Override
            public void success(Object response) {
                BaseResponse r = (BaseResponse)response;
                listener.onCreatedRoom(r.room);
            }

            @Override
            public void failure() {
                listener.onCreateFailed();
            }
        });
    }
}

File: app/src/main/kotlin/com/cyder/android/syncpod/api/request/CreateRoom.kt
package com.cyder.android.syncpod.api.request

/**
 * Created by chigichan24 on 2018/03/31.
 */

class CreateRoom(name: String, description: String, isPublic: Boolean) {
    private val room = Room(name, description, isPublic)
}

private data class Room (
        val name: String,
        val description: String,
        val public: Boolean
)